pytube/captions.py

```diff
@@ -65,32 +65,42 @@ def float_to_srt_time_format(d: float) -> str:
         ms = f"{fraction:.3f}".replace("0.", "")
         return time_fmt + ms
 
+
     def xml_caption_to_srt(self, xml_captions: str) -> str:
-        """Convert xml caption tracks to "SubRip Subtitle (srt)".
+      """Convert xml caption tracks to "SubRip Subtitle (srt)".
+
+      :param str xml_captions:
+          XML formatted caption tracks.
+      """
+      segments = []
+      root = ElementTree.fromstring(xml_captions)[1]
+      i=0
+      for child in list(root):
+          if child.tag == 'p':
+              caption = ''
+              if len(list(child))==0:
+                  continue
```

Comment:
```suggestion
                    if child.text:
                        caption += child.text
                    else:
                        continue
```
This, apparently, only works for auto-generated captions. 
But for the others, the text is directly in `child.text`